SpiceSharp/Simulations/Base/Frequency/ComplexSimulationState.cs

```diff
@@ -3,50 +3,93 @@
+            public Complex Laplace { get; set; } = new Complex();
 
-        /// <summary>
-        /// Gets or sets the current laplace variable.
-        /// </summary>
-        public Complex Laplace { get; set; } = new Complex();
+            /// <summary>
+            /// Gets the solver.
+            /// </summary>
+            /// <value>
+            /// The solver.
+            /// </value>
+            public ISparseSolver<Complex> Solver { get; }
 
-        /// <summary>
-        /// Setup the simulation state.
-        /// </summary>
-        /// <param name="nodes">The unknown variables for which the state is used.</param>
-        public override void Setup(VariableSet nodes)
-        {
-            nodes.ThrowIfNull(nameof(nodes));
-            Solution = new DenseVector<Complex>(Solver.Order);
-            base.Setup(nodes);
-        }
+            /// <summary>
+            /// Gets the variable to index map.
+            /// </summary>
+            /// <value>
+            /// The map.
+            /// </value>
+            public IVariableMap Map { get; private set; }
 
-        /// <summary>
-        /// Unsetup the state.
-        /// </summary>
-        public override void Unsetup()
-        {
-            Solution = null;
-            base.Unsetup();
+            /// <summary>
+            /// Initializes a new instance of the <see cref="ComplexSimulationState"/> class.
+            /// </summary>
+            public ComplexSimulationState()
+            {
+                Solver = LUHelper.CreateSparseComplexSolver();
+            }
+
+            /// <summary>
+            /// Initializes a new instance of the <see cref="ComplexSimulationState"/> class.
+            /// </summary>
+            /// <param name="solver">The solver.</param>
+            public ComplexSimulationState(ISparseSolver<Complex> solver)
+            {
+                Solver = solver.ThrowIfNull(nameof(solver));
+            }
+
+            /// <summary>
+            /// Initializes the specified simulation.
+            /// </summary>
+            /// <param name="simulation">The simulation.</param>
+            public void Initialize(ISimulation simulation)
+            {
+                simulation.ThrowIfNull(nameof(simulation));
+                Map = new VariableMap(simulation.Variables.Ground);
+            }
+
+            /// <summary>
+            /// Set up the simulation state for the simulation.
+            /// </summary>
+            /// <param name="simulation">The simulation.</param>
+            public void Setup(ISimulation simulation)
+            {
+                Solution = new DenseVector<Complex>(Solver.Size);
+            }
+
+            /// <summary>
+            /// Unsetup the state.
+            /// </summary>
+            public void Unsetup()
+            {
+                Solution = null;
+                Map.Clear();
+                Map = null;
+
+                // TODO: Clear all for the matrix
```

Comment:
@svenboulanger Is it still TODO?

Reply:
I believe the remark is about clearing the matrix elements (ie. making all elements eligible for garbage collection).

Reply:
Done